components/domains/registerSummary.tsx

```diff
@@ -5,56 +5,65 @@ import React, {
   useState,
 } from "react";
 import styles from "../../styles/components/registerV2.module.css";
-import CurrencySwitcher from "./currencySwitcher";
 import { gweiToEth, numberToFixedString } from "../../utils/feltService";
+import { CurrencyType } from "../../utils/constants";
+import CurrencyDropdown from "./currencyDropdown";
+import { Skeleton } from "@mui/material";
 
 type RegisterSummaryProps = {
   duration: number;
   ethRegistrationPrice: string;
+  registrationPrice: string; // price in displayedCurrency, set to priceInEth on first load as ETH is the default currency
   renewalBox: boolean;
   salesTaxRate: number;
   isSwissResident: boolean;
-  isUsdPriceDisplayed?: boolean;
+  isTokenDropdownDisplayed?: boolean;
   customMessage?: string;
+  displayedCurrency: CurrencyType;
+  onCurrencySwitch: (type: CurrencyType) => void;
+  loadingPrice?: boolean;
 };
 
 const RegisterSummary: FunctionComponent<RegisterSummaryProps> = ({
   duration,
   ethRegistrationPrice,
+  registrationPrice,
   renewalBox,
   salesTaxRate,
   isSwissResident,
-  isUsdPriceDisplayed = true,
+  isTokenDropdownDisplayed = true,
   customMessage,
+  displayedCurrency,
+  onCurrencySwitch,
+  loadingPrice,
 }) => {
-  const [isEthPriceDisplayed, setIsEthPriceDisplayed] = useState<boolean>(true);
-  const [ethSwissdPrice, setEthSwissdPrice] = useState<number>(0);
-  const [usdRegistrationPrice, setSwissdRegistrationPrice] =
-    useState<number>(0);
+  const [ethUsdPrice, setEthUsdPrice] = useState<number>(0); // price of 1ETH in USD
+  const [usdRegistrationPrice, setUsdRegistrationPrice] = useState<number>(0);
   const recurrence = renewalBox && duration === 1 ? "/year" : "";
   useEffect(() => {
     fetch(
       "https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd"
     )
       .then((res) => res.json())
       .then((data) => {
-        setEthSwissdPrice(data?.ethereum?.usd);
+        console.log("data", data);
+        setEthUsdPrice(data?.ethereum?.usd);
       })
       .catch((err) => console.log("Coingecko API Error:", err));
   }, []);
 
   useEffect(() => {
-    function computeSwissdPrice() {
-      if (ethSwissdPrice) {
-        return ethSwissdPrice * Number(gweiToEth(ethRegistrationPrice));
+    function computeUsdPrice() {
+      if (ethUsdPrice) {
+        return parseFloat(
```

Comment:
The reason is that you don't want to convert back a a formatted number to a float, it's not safe: sometimes you could end up with things buggy like 0.300000000...0004 instead of 0.3